src/pudl/transform/ferc1.py

```diff
@@ -2759,6 +2820,118 @@ def normalize_metadata_xbrl(
         return meta
 
 
+class BalanceSheetAssetsFerc1TableTransformer(Ferc1AbstractTableTransformer):
+    """Transformer class for :ref:`balance_sheet_assets_ferc1` table."""
+
+    table_id: TableIdFerc1 = TableIdFerc1.BALANCE_SHEET_ASSETS_FERC1
+    has_unique_record_ids: bool = False
+
+    def targeted_drop_duplicates_dbf(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Drop bad duplicate records from a specific utility in 1994.
+
+        This is a very specific fix, meant to get rid of a particular observed set of
+        duplicate records: FERC Respondent ID 165 in 1999 has two sets of assets and
+        other debt records, one of which contains one null record.
+
```

Comment:
hm maybe we want to make a more generic `target_drop` transformer... I could certainly imagine doing this but I think it would be hard to make it general enough and useful enough.

A lower hanging fruit option would be to make a no-op version of `targeted_drop_duplicates_dbf` within `Ferc1AbstractTableTransformer` that just takes a df and returns it within `process_dbf`. This way any table-transformer could override it with any bespoke needs and we could incorporate the three steps I added to this table's `process_dbf` to the generic version (I added `drop_invalid_rows`, `targeted_drop_duplicates_dbf` and `drop_duplicate_rows_dbf`).

We coooould also incorporate the dropping of records that *are* PK duplicates but have one (or more) null data value into `drop_duplicate_rows_dbf` to avoid the addition of `drop_invalid_rows` and `targeted_drop_duplicates_dbf`. I think I am partial to this, especially if you say your found this issue in your table as well.

Reply:
we went with this last suggestion: update `drop_duplicate_rows_dbf` so it drops dupes with *exactly* the same data as well as records with one null-having row and one data-full row.